ext/simplexml/simplexml.c

```diff
@@ -2378,11 +2384,14 @@ PHP_METHOD(SimpleXMLElement, __construct)
 	PHP_LIBXML_RESTORE_GLOBALS(read_file_or_memory);
 
 	if (!docp) {
-		((php_libxml_node_object *)sxe)->document = NULL;
 		zend_throw_exception(zend_ce_exception, "String could not be parsed as XML", 0);
 		RETURN_THROWS();
 	}
 
+	sxe_object_free_iterxpath(sxe);
+
+	php_libxml_decrement_node_ptr((php_libxml_node_object *)sxe);
```

Comment:
Isn't this done already by the `sxe_object_free_iterxpath` code?

Reply:
I do not think so

Reply:
in the freeing code you release the node, here you decrement it, I think just releasing should be enough.